### PR TITLE
[UX] Avoid known hosts check

### PR DESCRIPTION
**Pull request by Michaelvll**
A user reports that sometimes `ssh cluster-name` will complain about `REMOTE HOST IDENTIFICATION HAS CHANGED` error.
image

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c min --cloud gcp; ssh min`

### CODE REVIEW

File: sky/backends/backend_utils.py [hunk 1]
```diff
@@ -380,6 +384,7 @@ def _get_generated_config(cls, autogen_comment: str, host_name: str,
               IdentitiesOnly yes
               ForwardAgent yes
               StrictHostKeyChecking no
+              UserKnownHostsFile=/dev/null
```
Comment (romilbhardwaj) [inline sky/backends/backend_utils.py:387]: Curious - do we also need to set `GlobalKnownHostsFile=/dev/null`? [[1]](https://man.openbsd.org/ssh_config#GlobalKnownHostsFile) [[2]](https://www.baeldung.com/linux/ssh-known_hosts-ignore-emporarily#3-permanently-ignore)

File: sky/backends/backend_utils.py [hunk 2]
```diff
@@ -371,6 +371,10 @@ def _get_generated_config(cls, autogen_comment: str, host_name: str,
             proxy = f'ProxyCommand {proxy_command}'
         else:
             proxy = ''
+        # StrictHostKeyChecking=no skips the host key check for the first
+        # time. UserKnownHostsFile=/dev/null prevents the host key from being
+        # added to the known_hosts file, so that the host key check is skipped
+        # for all subsequent connections.
```
Comment (romilbhardwaj) [inline sky/backends/backend_utils.py:377]: In addition to not allowing adding of the host_key, I think setting `UserKnownHostsFile=/dev/null` also always returns an empty file for known hosts, thus making SSH think this is a first-time connection and thus skipping the host key check. Might be worth mentioning in the comment

**Review by romilbhardwaj — COMMENTED** [review]
Thanks for adding this @Michaelvll!

File: sky/backends/backend_utils.py (hunk 1 above)
Comment (Michaelvll) [reply sky/backends/backend_utils.py:389]: Ahh, good catch! It seems that `GlobalKnownHostsFile=/dev/null` should be included as well. Thanks!

File: sky/backends/backend_utils.py (hunk 2 above)
Comment (Michaelvll) [reply sky/backends/backend_utils.py]: Good point! Added. Thanks!

**Review by romilbhardwaj — APPROVED** [review]
lgtm, thanks @Michaelvll!

**Pull request merged**